tests/TestCase/Datasource/ConnectionManagerTest.php

```diff
@@ -306,6 +306,31 @@ public function dsnProvider()
                     'log' => '1'
                 ]
             ],
+            'no password' => [
+                'mysql://user@localhost:3306/database',
```

Comment:
I misunderstood the behavior a little bit. The related code is:
https://github.com/cakephp/cakephp/blob/8a91fca6b147ebc0eb39c77ef330e357207dfd5a/src/Core/StaticConfigTrait.php#L96-L100

So, probably, people could use a config array like this:
```php
[
    'password' => env('DATABASE_PASSWORD'),
    'url' => 'mysql://user@localhost:3306/database',
]
```